fhir-model/src/test/java/com/ibm/fhir/model/test/ResourceCoverageTest.java

```diff
@@ -0,0 +1,260 @@
+                        assertNotNull(resource2);
+                        assertTrue(true);
+                    }
+                }
+                
+                try (StringWriter writer = new StringWriter()) {
+                    FHIRGenerator.generator(Format.XML).generate(resource, writer);
+                    String outXML = writer.toString();
+                    assertNotNull(outXML);
+                    assertFalse(outXML.isEmpty());
+                    
+                    try (ByteArrayInputStream in = new ByteArrayInputStream(outXML.getBytes())) {
+                        Resource resource2 = FHIRParser.parser(Format.XML).parse(in).as(Resource.class);
+                        assertNotNull(resource2);
+                        assertTrue(true);
+                    }
+                }
+            }
+        }
+    }
+
+    public void runMethods(Object resource, Method[] methods) throws Exception {
+        for (Method method : methods) {
+            if (method.getName().startsWith("get")) {
+                method.invoke(resource);
+                assertTrue(true);
+            }
+
+            if (method.getName().equals("hashCode")) {
+                method.invoke(resource);
+                assertTrue(true);
+            }
+
+            if (method.getName().equals("equals")) {
+                method.invoke(resource, resource);
+                assertTrue(true);
+
+                String x = "BAD";
+                method.invoke(resource, x);
+                assertTrue(true);
+                
+                if (resource instanceof Resource) {
+                    Resource r = (Resource) resource; 
+                    Resource r2 = r.toBuilder().build(); 
+                    assertTrue(r.equals(r2));
+                }
+            }
+
+            if (method.getName().equals("hasChildren")) {
+                method.invoke(resource);
+                assertTrue(true);
+            }
+        }
+    }
+
+    public void buildTestForUncoveredTypes() throws Exception {
+        List<Extension> extensions = new ArrayList<>();
+        extensions.add(buildBase64Binary());
+        extensions.add(buildBooleanTrue());
+        extensions.add(buildCanonical());
+        extensions.add(buildId());
+        extensions.add(buildContactDetail());
+        extensions.add(buildCount());
+        extensions.add(buildDistance());
+        extensions.add(buildMoneyQuantity());
+
+        CodeableConcept code = CodeableConcept.builder().id("1-2-3-4").text(string("Demo")).build();
+        Basic basic = Basic.builder().code(code).extension(extensions).build();
+
+        try (StringWriter writer = new StringWriter()) {
+            FHIRGenerator.generator(Format.XML).generate(basic, writer);
+
+            String out = writer.toString();
+            assertNotNull(out);
+            assertFalse(out.isEmpty());
+            System.out.println(out);
```

Comment:
do we really want to print them out always?

Reply:
this only builds a test.  The corresponding content is in fhir-examples